src/pubkey.rs

```diff
@@ -272,17 +272,48 @@ impl PublicKey {
     /// assert_eq!(fp.hash, "ciQkdxjFUhk2E2vRkWJD9kB8pi+EneOkaCJJHNWzPC4");
     /// ```
     pub fn from_string(contents: &str) -> Result<PublicKey> {
-        let mut iter = contents.split_whitespace();
-
-        let kt_name = iter
-            .next()
-            .ok_or(Error::with_kind(ErrorKind::InvalidFormat))?;
-
-        let data = iter
+        // Ssh public keys are on a single line, so we need to remove any new lines now.
+        let contents = contents
+            .lines()
             .next()
             .ok_or(Error::with_kind(ErrorKind::InvalidFormat))?;
 
-        let comment = iter.next().map(|v| String::from(v));
+        // Per man 8 sshd:
+        // """
+        // The optional comment field continues to the end of the line
+        // """
+        //
+        // This means we can't simply use str::split_whitespace since that can eat
+        // the comment field if the comment has spaces (which are valid).
+        // Additionally the function we *need* is currently nightly only.
+        //
+        // https://doc.rust-lang.org/std/str/struct.SplitWhitespace.html#method.remainder
+        //
+        // So this means we have to do this the hard way - we need to walk the string
+        // finding and splitting as we go. That's what the helper "split_whitespace" does
+        // for us.
+        let (kt_name, remainder) =
+            split_whitespace(contents).ok_or(Error::with_kind(ErrorKind::InvalidFormat))?;
+
+        // If we have the ability to split, this means we have a comment.
+        // If the split fails, this means we only have data.
+        let (data, comment) = split_whitespace(remainder)
+            .map(|(data, comment)| {
+                // To keep this somewhat reasonable, we limit the comment to 256 chars.
```

Comment:
I'm not aware of such limit in the ssh key comment. Can we extend this to consume the comment to the end of the line?